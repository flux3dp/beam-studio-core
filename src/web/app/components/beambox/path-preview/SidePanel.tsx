import classNames from 'classnames';
import React from 'react';

import i18n from 'helpers/i18n';

interface Props {
  size: string;
  estTime: string;
  lightTime: string;
  rapidTime: string;
  cutDist: string;
  rapidDist: string;
  currentPosition: string;
  handleStartHere: () => void;
  isStartHereEnabled: boolean;
  togglePathPreview: () => void;
}

function SidePanel({
  size,
  estTime,
  lightTime,
  rapidTime,
  cutDist,
  rapidDist,
  currentPosition,
  handleStartHere,
  isStartHereEnabled,
  togglePathPreview,
}: Props): JSX.Element {
  const LANG = i18n.lang.beambox.path_preview;
  const renderDataBlock = (label: string, value: string): JSX.Element => (
    <div className="data-block">
      <div className="item">{label}</div>
      <div className="value">{value}</div>
    </div>
  );

  return (
    <div id="path-preview-side-panel" className={classNames({ win: window.os === 'Windows' })}>
      <div className="title">{LANG.preview_info}</div>
      <div className="datas">
        {renderDataBlock(LANG.size, size)}
        {renderDataBlock(LANG.estimated_time, estTime)}
        {renderDataBlock(LANG.cut_time, lightTime)}
        {renderDataBlock(LANG.rapid_time, rapidTime)}
        {renderDataBlock(LANG.cut_distance, cutDist)}
        {renderDataBlock(LANG.rapid_distance, rapidDist)}
        {renderDataBlock(LANG.current_position, currentPosition)}
      </div>
      <div className="remark">
        {LANG.remark}
      </div>
      <div className="buttons">
        <div
          className={classNames('btn btn-default primary', { disabled: !isStartHereEnabled })}
          onClick={isStartHereEnabled ? handleStartHere : null}
        >
          {LANG.start_here}
        </div>
        <div className="btn btn-default" onClick={togglePathPreview}>
          {LANG.end_preview}
        </div>
      </div>
    </div>
  );
}

export default SidePanel;
