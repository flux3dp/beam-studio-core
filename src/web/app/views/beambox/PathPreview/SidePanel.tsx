import classNames from 'classnames';
import React from 'react';

interface Props {
  size: string;
  estTime: string;
  lightTime: string;
  rapidTime: string;
  cutDist: string;
  rapidDist: string;
  currentPosition: string;
  handleStartHere: () => void;
  togglePathPreview: () => void;
}

class SidePanel extends React.PureComponent<Props> {
  // eslint-disable-next-line class-methods-use-this
  renderDataBlock(label: string, value: string): JSX.Element {
    return (
      <div className="data-block">
        <div className="item">{label}</div>
        <div className="value">{value}</div>
      </div>
    );
  }

  render(): JSX.Element {
    const className = classNames({ win: window.os === 'Windows' });
    const {
      size, estTime, lightTime, rapidTime, cutDist, rapidDist, currentPosition,
      handleStartHere, togglePathPreview,
    } = this.props;

    return (
      <div id="path-preview-side-panel" className={className}>
        <div className="title">Preview Data</div>
        <div className="datas">
          {this.renderDataBlock('Size', size)}
          {this.renderDataBlock('Estimated Time', estTime)}
          {this.renderDataBlock('Light Time', lightTime)}
          {this.renderDataBlock('Rapid Time', rapidTime)}
          {this.renderDataBlock('Cut Distance', cutDist)}
          {this.renderDataBlock('Rapid Distance', rapidDist)}
          {this.renderDataBlock('Current Position', currentPosition)}
        </div>
        <div className="buttons">
          <div className="btn btn-default primary" onClick={handleStartHere}>
            Start Here
          </div>
          <div className="btn btn-default" onClick={togglePathPreview}>
            End Preview
          </div>
        </div>
      </div>
    );
  }
}

export default SidePanel;
