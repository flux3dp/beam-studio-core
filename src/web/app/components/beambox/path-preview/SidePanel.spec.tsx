/* eslint-disable import/first */
import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

jest.mock('helpers/i18n', () => ({
  lang: {
    beambox: {
      path_preview: {
        play: 'Play',
        pause: 'Pause',
        stop: 'Stop',
        play_speed: 'Playback Speed',
        travel_path: 'Travel Path',
        invert: 'Invert',
        preview_info: 'Preview Information',
        size: 'Size',
        estimated_time: 'Total Time Estimated',
        cut_time: 'Cut Time',
        rapid_time: 'Travel Time',
        cut_distance: 'Cut Distance',
        rapid_distance: 'Travel Distance',
        current_position: 'Current Position',
        remark: '* All pieces of information are estimated value for reference.',
        start_here: 'Start Here',
        end_preview: 'End Preview',
      },
    },
  },
}));

import SidePanel from './SidePanel';

describe('side panel test', () => {
  it('should render correctly when enabled', () => {
    const handleStartHere = jest.fn();
    const togglePathPreview = jest.fn();
    const wrapper = shallow(<SidePanel
      size="100 x 100 mm"
      estTime="60 s"
      lightTime="30 s"
      rapidTime="10 s"
      cutDist="50 mm"
      rapidDist="30 mm"
      currentPosition="50, 50 mm"
      handleStartHere={handleStartHere}
      isStartHereEnabled
      togglePathPreview={togglePathPreview}
    />);
    expect(toJson(wrapper)).toMatchSnapshot();

    wrapper.find('div.btn-default').at(0).simulate('click');
    expect(handleStartHere).toHaveBeenCalledTimes(1);

    wrapper.find('div.btn-default').at(1).simulate('click');
    expect(togglePathPreview).toHaveBeenCalledTimes(1);
  });

  it('should render correctly when disabled', () => {
    const handleStartHere = jest.fn();
    const togglePathPreview = jest.fn();
    const wrapper = shallow(<SidePanel
      size="100 x 100 mm"
      estTime="60 s"
      lightTime="30 s"
      rapidTime="10 s"
      cutDist="50 mm"
      rapidDist="30 mm"
      currentPosition="50, 50 mm"
      handleStartHere={handleStartHere}
      isStartHereEnabled={false}
      togglePathPreview={togglePathPreview}
    />);
    expect(toJson(wrapper)).toMatchSnapshot();

    wrapper.find('div.btn-default').at(0).simulate('click');
    expect(handleStartHere).toHaveBeenCalledTimes(0);

    wrapper.find('div.btn-default').at(1).simulate('click');
    expect(togglePathPreview).toHaveBeenCalledTimes(1);
  });
});
