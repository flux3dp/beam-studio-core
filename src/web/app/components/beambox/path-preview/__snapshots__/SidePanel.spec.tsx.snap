// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className=""
  id="path-preview-side-panel"
>
  <div
    className="title"
  >
    Preview Information
  </div>
  <div
    className="datas"
  >
    <div
      className="data-block"
    >
      <div
        className="item"
      >
        Size
      </div>
      <div
        className="value"
      >
        100 x 100 mm
      </div>
    </div>
    <div
      className="data-block"
    >
      <div
        className="item"
      >
        Total Time Estimated
      </div>
      <div
        className="value"
      >
        60 s
      </div>
    </div>
    <div
      className="data-block"
    >
      <div
        className="item"
      >
        Cut Time
      </div>
      <div
        className="value"
      >
        30 s
      </div>
    </div>
    <div
      className="data-block"
    >
      <div
        className="item"
      >
        Travel Time
      </div>
      <div
        className="value"
      >
        10 s
      </div>
    </div>
    <div
      className="data-block"
    >
      <div
        className="item"
      >
        Cut Distance
      </div>
      <div
        className="value"
      >
        50 mm
      </div>
    </div>
    <div
      className="data-block"
    >
      <div
        className="item"
      >
        Travel Distance
      </div>
      <div
        className="value"
      >
        30 mm
      </div>
    </div>
    <div
      className="data-block"
    >
      <div
        className="item"
      >
        Current Position
      </div>
      <div
        className="value"
      >
        50, 50 mm
      </div>
    </div>
  </div>
  <div
    className="remark"
  >
    * All pieces of information are estimated value for reference.
  </div>
  <div
    className="buttons"
  >
    <div
      className="btn btn-default primary"
      onClick={[MockFunction]}
    >
      Start Here
    </div>
    <div
      className="btn btn-default"
      onClick={[MockFunction]}
    >
      End Preview
    </div>
  </div>
</div>
`;
